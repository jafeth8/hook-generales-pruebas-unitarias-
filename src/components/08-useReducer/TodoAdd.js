import React from 'react'
import { UseForm } from '../../hooks/UseForm';

export const TodoAdd = ({handleAddTodo}) => {
    const [{description}, handleChange, reset]=UseForm({description:''});

    const handleSubmit=(e)=> {
        e.preventDefault();

        if(description.trim().length<3){
            return;
        }
        
        const newTodo = {
            id:new Date().getTime(),
            desc:description,
            done: false
        }

        handleAddTodo(newTodo);
        reset();
    }

    return (
        <>
            <h4>Agregar TODO</h4>
            <hr/>

            <form onSubmit={handleSubmit}>
                <input type="text" name="description" 
                placeholder="Aprender..." autoComplete="off" 
                className="form-control" value={description}
                onChange={handleChange}
                />

                <button className="btn btn-outline-primary mt-1 w-100" type="submit">
                    Agregar
                </button>
            </form>     
        </>
    )
}
