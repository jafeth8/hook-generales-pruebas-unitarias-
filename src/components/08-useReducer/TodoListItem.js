import React from 'react'
import PropTypes from 'prop-types'


export const TodoListItem = ({todo,index,handleToggle,handleDelete}) => {
    return (
        <li className="list-group-item">
            <p onClick={()=>handleToggle(todo.id)} className={ todo.done? "complete" : "" } >{index+1}. {todo.desc}</p>
            <button className="btn btn-danger" onClick={()=>handleDelete(todo.id)}>borrar</button>
        </li>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired,
}


