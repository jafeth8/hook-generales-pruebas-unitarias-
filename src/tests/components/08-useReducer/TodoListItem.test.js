import {shallow} from 'enzyme';
import { TodoListItem } from '../../../components/08-useReducer/TodoListItem';
import {demoTodos} from '../../fixtures/demoTodos'


describe('pruebas en <TodoListItem/>', () => {
    
    const handleToggle=jest.fn()
    const handleDelete=jest.fn();

    const wrapper =shallow(<TodoListItem todo={demoTodos[0]} index={0} handleToggle={handleToggle} handleDelete={handleDelete}/>); 

    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        
    })

    test('debe de llamar la funcion handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalled();
        expect(handleDelete).toHaveBeenCalledWith(123);//se llamo con el id del TODO
    })

    test('debe de llamar la funcion handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalled();
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);//se cambio el estado con el id del TODO (otra forma de evaluar el id 123 como en la prueba anterior-- tambien es valida)
    })
    
    test('debe de mostrar el texto correctamente', () => {
        const texto =wrapper.find('p').text().trim()
        expect(texto).toBe('1. Aprender laravel')
    })
    
    test('la etiqueta <p></p> debe de tener la clase complete si el TODO.done=true', () => {
        const todo=demoTodos[0];
        todo.done=true;

        const wrapper =shallow(
        <TodoListItem 
        todo={todo} 
        index={0} 
        handleToggle={handleToggle} 
        handleDelete={handleDelete}/>); 
        
        expect(wrapper.find('p').hasClass('complete')).toBe(true);
        
        expect(wrapper.find('.complete').exists()).toBe(true);//otra forma de evaluarlo
        
    })
    
    
    
})
