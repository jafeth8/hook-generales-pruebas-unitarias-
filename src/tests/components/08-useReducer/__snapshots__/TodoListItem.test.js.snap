// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoListItem/> debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    . 
    Aprender laravel
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    borrar
  </button>
</li>
`;
